lib/server/services/run/RunService.js

```diff
@@ -275,6 +277,28 @@ class RunService {
             queryBuilder.include('lhcPeriod');
         }
     }
+
+    /**
+     * Calculate distinct combination of levels of alice L3 and dipole current rounded to kilo amperes
+     *
+     * @return {{l3Level: number, dipoleLevel: number}[]} distinct combination list
+     */
+    async getAllAliceL3AndDipoleLevelsForPhysicsRuns() {
+        const queryBuilder = dataSource.createQueryBuilder()
+            .set('attributes', (sequelize) => [
+                [sequelize.literal('DISTINCT ROUND(IF(alice_l3_polarity = \'NEGATIVE\', -1, 1) * alice_l3_current / 1000)'), 'l3Level'],
+                [sequelize.literal('ROUND(IF(alice_dipole_polarity = \'NEGATIVE\', -1, 1) * alice_dipole_current / 1000)'), 'dipoleLevel'],
+            ])
+            .where('aliceL3Current').not().is(null)
+            .where('aliceL3Polarity').not().is(null)
+            .where('aliceDipoleCurrent').not().is(null)
+            .where('aliceDipolePolarity').not().is(null)
+            .where('definition').is(RunDefinition.Physics);
```

Comment:
I think we also want to filter on good.
We want all the combinations, you should probably use the `group by`

As reminder, the request that gave me good results: 
```sql
SELECT DISTINCT CONCAT(
                    IF(alice_l3_polarity = 'NEGATIVE' AND ROUND(alice_l3_current / 1000) <> 0, '-', ''),
                    ROUND(alice_l3_current / 1000),
                    'kA/',
                    IF(alice_dipole_polarity = 'NEGATIVE' AND ROUND(alice_dipole_current / 1000) <> 0, '-', ''),
                    ROUND(alice_dipole_current / 1000),
                    'kA'
                )
FROM runs
WHERE alice_l3_current IS NOT NULL
  AND alice_l3_polarity IS NOT NULL
  AND alice_dipole_current IS NOT NULL
  AND alice_dipole_polarity IS NOT NULL
  AND definition = 'PHYSICS'
  AND run_quality = 'good'
ORDER BY alice_l3_polarity, alice_l3_current, alice_dipole_current, alice_dipole_current;
```

Reply:
not on good, as data passes can have bad runs. There are all combinations, `group by` doesn't work here there is `distinct`